okhttp-android/src/androidTest/kotlin/okhttp3/android/AndroidAsyncDnsTest.kt

```diff
@@ -94,7 +95,7 @@ class AndroidAsyncDnsTest(val server: MockWebServer) {
 
   @Test
   fun testRequestInvalid() {
-    val call = client.newCall(Request.Builder().url("https://google.invalid/").build())
+    val call = client.newCall(Request("https://google.invalid/".toHttpUrl()))
```

Comment:
I think I know the answer here...

But what about

```
"https://google.invalid/".toHttpUrl().toRequest()
"https://google.invalid/".toHttpUrl().toRequest(method="POST")
```

with a default for method?

Reply:
It feels just slightly like a layering violation. I don't think `.toXxx()` methods are as obvious when they take ‘important’ parameters?

(I can't necessary defend this position while justifying `toHttpUrl()` and `toRequestBody()`, but this one feels off.)